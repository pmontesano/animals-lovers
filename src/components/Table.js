import React from 'react';

const Table = ({ users, photo, handleUserClick }) => {
  const handleButtonRemove = (id) => (e) => {
    handleUserClick(id);
  };

  return (
    <div className="flex flex-col">
      <div className="-my-2 overflow-x-auto sm:-mx-6 lg:-mx-8">
        <div className="py-2 align-middle inline-block min-w-full sm:px-6 lg:px-8 overflow-x-auto">
          <div className="shadow overflow-hidden border-b border-gray-200 sm:rounded-lg overflow-x-auto">
            <table className="min-w-full divide-y divide-gray-200">
              <thead className="bg-gray-50">
                <tr>
                  <th
                    scope="col"
                    className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
                  >
                    Name
                  </th>
                  <th
                    scope="col"
                    className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
                  >
                    Animal List
                  </th>
                  <th
                    scope="col"
                    className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
                  >
                    Points
                  </th>
                  <th
                    scope="col"
                    className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
                  >
                    Actions
                  </th>
                </tr>
              </thead>
              <tbody className="bg-white divide-y divide-gray-200">
                {users.map((user) => (
                  <tr key={user.id}>
                    <td className="px-6 py-4 whitespace-nowrap">
                      <div className="flex items-center">
                        <div className="flex-shrink-0 h-10 w-10">
                          <img
                            className="h-10 w-10 rounded-full"
                            src={
                              photo
                                ? photo
                                : 'https://images.unsplash.com/photo-1491528323818-fdd1faba62cc?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80'
                            }
                            alt={user.firstName}
                          />
                        </div>
                        <div className="ml-4">
                          <div className="text-sm font-medium text-gray-900">
                            {user.firstName} {user.lastName}
                          </div>
                        </div>
                      </div>
                    </td>
                    <td className="px-6 py-4 whitespace-nowrap">
                      <ul className="inline-list">
                        {user.animals.map((animal, i) => (
                          <li key={i} className="text-gray-400">
                            {' '}
                            {animal}{' '}
                          </li>
                        ))}
                      </ul>
                    </td>
                    <td className="px-6 py-4 whitespace-nowrap text-sm text-yellow-500 text-lg">
                      {user.points}
                    </td>
                    <td className="px-6 py-4 whitespace-nowrap text-right text-sm font-medium">
                      <button
                        className="btn-remove text-blue-400 hover:text-blue-500"
                        onClick={handleButtonRemove(user.id)}
                      >
                        Delete user
                      </button>
                    </td>
                  </tr>
                ))}
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Table;
