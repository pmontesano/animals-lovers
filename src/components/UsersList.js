import Table from './Table';

const UsersList = ({
  animalName,
  users,
  handleClick,
  handleRemoveUserClick,
  fullList,
}) => {
  const limit = 25;

  const handleUserClick = (id) => {
    handleRemoveUserClick(id);
  };

  const handleShowMoreClick = (limit) => (e) => {
    handleClick(limit);
  };

  const title = animalName.toUpperCase() + ' USERS';

  return (
    <div className="max-w-7xl mx-auto sm:px-6 lg:px-8">
      <h2 className="font-medium text-2xl py-6">{title}</h2>
      <Table
        users={users}
        handleUserClick={handleUserClick}
        thElements={['Name', 'Animal List', 'Point', 'Actions']}
      />
      {!fullList && (
        <div className="text-center py-8">
          <button
            className="btn btn-blue btn-blue:hover"
            onClick={handleShowMoreClick(limit)}
          >
            Show more
          </button>
        </div>
      )}
    </div>
  );
};

export default UsersList;
