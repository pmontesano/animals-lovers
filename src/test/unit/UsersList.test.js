import React from 'react';
import { create } from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import UsersList from '../../components/UsersList';
import Table from '../../components/Table';
import shallow from 'enzyme/build/shallow';

configure({ adapter: new Adapter() });

const handleClickButton = jest.fn();
const handleShowMoreClick = jest.fn();

const props = {
  animalName: 'Bear',
  users: [
    {
      age: 33,
      animals: [],
      firstName: 'Stefanie',
      id: '5fbfe2a4aecbbb9031030578',
      lastName: 'Bond',
      points: 100,
    },
  ],
  handleRemoveUserClick: handleClickButton,
  fullList: false,
  handleClick: handleShowMoreClick,
};

describe('UsersList', () => {
  it('should render correctly', () => {
    const wrapper = create(<UsersList {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render Table component inside', () => {
    const wrapper = shallow(<UsersList {...props} />);

    expect(wrapper.find(Table)).toHaveLength(1);
  });

  it('should click handleShowMoreClick', () => {
    const wrapper = shallow(<UsersList {...props} />);

    // is the limit of users's list when the handleShowMoreClick is clicked
    const maxLimit = 25;

    wrapper.find('button').simulate('click');
    expect(handleShowMoreClick.mock.calls).toEqual([[maxLimit]]);
  });
});
