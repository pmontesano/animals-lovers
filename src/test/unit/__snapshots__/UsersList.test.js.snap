// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersList should render correctly 1`] = `
<div
  className="max-w-7xl mx-auto sm:px-6 lg:px-8"
>
  <h2
    className="font-medium text-2xl py-6"
  >
    BEAR USERS
  </h2>
  <div
    className="flex flex-col"
  >
    <div
      className="-my-2 overflow-x-auto sm:-mx-6 lg:-mx-8"
    >
      <div
        className="py-2 align-middle inline-block min-w-full sm:px-6 lg:px-8"
      >
        <div
          className="shadow overflow-hidden border-b border-gray-200 sm:rounded-lg"
        >
          <table
            className="min-w-full divide-y divide-gray-200"
          >
            <thead
              className="bg-gray-50"
            >
              <tr>
                <th
                  className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
                  scope="col"
                >
                  Name
                </th>
                <th
                  className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
                  scope="col"
                >
                  Animal List
                </th>
                <th
                  className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
                  scope="col"
                >
                  Points
                </th>
                <th
                  className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
                  scope="col"
                >
                  Actions
                </th>
              </tr>
            </thead>
            <tbody
              className="bg-white divide-y divide-gray-200"
            >
              <tr>
                <td
                  className="px-6 py-4 whitespace-nowrap"
                >
                  <div
                    className="flex items-center"
                  >
                    <div
                      className="flex-shrink-0 h-10 w-10"
                    >
                      <img
                        alt="Stefanie"
                        className="h-10 w-10 rounded-full"
                        src="https://images.unsplash.com/photo-1491528323818-fdd1faba62cc?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80"
                      />
                    </div>
                    <div
                      className="ml-4"
                    >
                      <div
                        className="text-sm font-medium text-gray-900"
                      >
                        Stefanie
                         
                        Bond
                      </div>
                    </div>
                  </div>
                </td>
                <td
                  className="px-6 py-4 whitespace-nowrap"
                >
                  <ul
                    className="inline-list"
                  />
                </td>
                <td
                  className="px-6 py-4 whitespace-nowrap text-sm text-gray-500"
                >
                  100
                </td>
                <td
                  className="px-6 py-4 whitespace-nowrap text-right text-sm font-medium"
                >
                  <button
                    className="btn-remove-user"
                    onClick={[Function]}
                  >
                    Delete user
                  </button>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
  <div
    className="text-center py-8"
  >
    <button
      className="btn btn-blue btn-blue:hover"
      onClick={[Function]}
    >
      Show more
    </button>
  </div>
</div>
`;
